test/client-utils.js

```diff
@@ -104,37 +105,44 @@ async function joinRoom({ client, roomId, viaServers }) {
   return joinedRoomId;
 }
 
-async function sendEvent({ client, roomId, eventType, content, timestamp }) {
+async function sendEvent({ client, roomId, eventType, stateKey, content, timestamp }) {
   assert(client);
   assert(roomId);
   assert(content);
 
   let qs = new URLSearchParams();
   if (timestamp) {
     assert(
-      timestamp && client.userId,
+      timestamp && client.applicationServiceUserIdOverride,
       'We can only do `?ts` massaging from an application service access token. ' +
-        'Expected `client.userId` to be defined so we can act on behalf of that user'
+        'Expected `client.applicationServiceUserIdOverride` to be defined so we can act on behalf of that user'
     );
 
     qs.append('ts', timestamp);
   }
 
-  if (client.userId) {
-    qs.append('user_id', client.userId);
+  if (client.applicationServiceUserIdOverride) {
+    qs.append('user_id', client.applicationServiceUserIdOverride);
   }
 
-  const sendResponse = await fetchEndpointAsJson(
-    urlJoin(
+  let url;
+  if (stateKey) {
+    url = urlJoin(
+      client.homeserverUrl,
+      `/_matrix/client/v3/rooms/${roomId}/state/${eventType}/${stateKey}?${qs.toString()}`
+    );
+  } else {
```

Comment:
Added support to send state events but ended up using the dedicated `/profile` endpoints in their own `updateProfile` function